yt_dlp/postprocessor/ffmpeg.py

```diff
@@ -614,57 +617,87 @@ def run(self, info):
-            if not mp4_ass_warn and ext == 'mp4' and sub_ext == 'ass':
-                mp4_ass_warn = True
-                self.report_warning('ASS subtitles cannot be properly embedded in mp4 files; expect issues')
-
-        if not sub_langs:
+                if ext == 'mp4' and sub_info['ext'] == 'ass':
+                    warn_once('ASS subtitles cannot be properly embedded in mp4 files; expect issues')
+                subtitles_to_embed[lang] = sub_info
+
+        if not subtitles_to_embed:
             return [], info
 
-        input_files = [filename] + sub_filenames
+        sub_files = [sub['filepath'] for sub in subtitles_to_embed.values()]
+        files_to_delete = [] if self._already_have_subtitle else sub_files
+
+        if ext in self.SUPPORTS_LYRICS:
+            self._embed_lyrics(subtitles_to_embed, info['filepath'], ext)
+            return files_to_delete, info
 
         opts = [
-            *self.stream_copy_opts(ext=info['ext']),
+            *self.stream_copy_opts(ext=ext),
             # Don't copy the existing subtitles, we may be running the
             # postprocessor a second time
             '-map', '-0:s',
         ]
-        for i, (lang, name) in enumerate(zip(sub_langs, sub_names)):
-            opts.extend(['-map', '%d:0' % (i + 1)])
-            lang_code = ISO639Utils.short2long(lang) or lang
-            opts.extend(['-metadata:s:s:%d' % i, 'language=%s' % lang_code])
-            if name:
-                opts.extend(['-metadata:s:s:%d' % i, 'handler_name=%s' % name,
-                             '-metadata:s:s:%d' % i, 'title=%s' % name])
+        for i, (lang, sub) in enumerate(subtitles_to_embed.items()):
+            lang = ISO639Utils.short2long(lang) or lang
+            opts.extend(['-map', f'{i + 1}:0', f'-metadata:s:s:{i}', f'language={lang}'])
+            if name := sub['name']:
+                opts.extend([f'-metadata:s:s:{i}', f'handler_name={name}',
+                             f'-metadata:s:s:{i}', f'title={name}'])
 
         temp_filename = prepend_extension(filename, 'temp')
-        self.to_screen('Embedding subtitles in "%s"' % filename)
-        self.run_ffmpeg_multiple_files(input_files, temp_filename, opts)
+        self.to_screen(f'Embedding subtitles in "{filename}"')
+        self.run_ffmpeg_multiple_files([filename, *sub_files], temp_filename, opts)
         os.replace(temp_filename, filename)
 
-        files_to_delete = [] if self._already_have_subtitle else sub_filenames
         return files_to_delete, info
 
+    def _embed_lyrics(self, subtitles, filename, ext):
+        assert mutagen and ext in self.SUPPORTS_LYRICS and all(sub['ext'] == 'lrc' for sub in subtitles.values())
+        self.to_screen(f'Embedding lyrics in "{filename}"')
+        if len(subtitles) > 1:
+            self.report_warning(
+                f'Your media player may be unable to display multiple subtitles in {ext}')
```

Comment:
maybe this should be `only_once`?

Reply:
should i do it?

pr has been open for too long, is there any other pr that is holding it?